woocommerce/woocommerce
# Product Collection: Return to top after page change

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces a "Scroll to Top" feature upon page change. It also involves a significant refactor in how we utilize the Interactivity API for client-side pagination i.e. the transition from using the `navigation-link` directive to `actions`.

Closes #42195 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new post & Add Product Collection block
2. Select “Product Catalog” collection
3. Publish the changes & go to Frontend.
4. After changing pages, confirm that the page scrolls to the top of the Product Collection block, as illustrated in the GIF below:
![Screen Capture on 2024-01-23 at 12-21-57 pm](https://github.com/woocommerce/woocommerce/assets/16707866/49c38891-818b-42a4-b528-75b8fd4e0624)

5. In the Editor, adjust the “ITEMS PER PAGE” setting in the toolbar controls to 3.
6. Save these changes and ensure that pagination functions correctly on the Frontend.
    1. Experiment with changing pages rapidly and check that the pagination block appears and operates as expected.
7. Now add multiple Product Collection to same post & verify that:
    1. Pagination works as expected.
    2. Confirm that the "Scroll to top" feature operates properly for the Product Collection block when you change its page.

Here is a quick demo video:

https://github.com/woocommerce/woocommerce/assets/16707866/099326c0-7ea6-447d-b41b-bddac470cbe3



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Product Collection: Return to top after the page change

#### Comment 

</details>
